test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_core_test.clj

```diff
@@ -174,3 +177,126 @@
+        ;; and I didn't see any other way to pull the queue back from
+        ;; the thread pool.
+        get-queue-method (-> thread-pool
+                             (.getClass)
+                             (.getDeclaredMethod "getQueue" nil))
+        _                (.setAccessible get-queue-method true)
+        queue            (.invoke get-queue-method thread-pool nil)]
+    {:thread-pool thread-pool
+     :queue queue}))
+
+(deftest create-server-tests
+  (testing "thread pool given proper values"
+    (testing "max threads passed through"
+      (dotimes [x 5]
+        (let [{:keys [thread-pool]}
+          (get-thread-pool-and-queue-for-create-server x 1)]
+          (is (= x (.getMaxThreads thread-pool))
+              "Unexpected max threads for queue"))))
+
+    (testing "default idle timeout passed through"
+      (let [{:keys [thread-pool]}
+             (get-thread-pool-and-queue-for-create-server 42 1)]
+        (is (= jetty/default-queue-idle-timeout
+               (.getIdleTimeout thread-pool)))))
+
+    (testing "when queue-max-size less than default-queue-min-threads"
+      (let [max-threads    23
+            queue-min-size (dec jetty/default-queue-min-threads)
+            {:keys [thread-pool queue]}
+              (get-thread-pool-and-queue-for-create-server max-threads
+                                                           queue-min-size)]
+        (is (= max-threads (.getMaxThreads thread-pool))
+            "Unexpected max threads for thread pool")
+        (is (= queue-min-size (.getMinThreads thread-pool))
+            "Unexpected min threads for queue")
+        (is (= queue-min-size (.getCapacity queue))
+            "Unexpected initial capacity for queue")
+        (is (= queue-min-size (.getMaxCapacity queue))
+            "Unexpected max capacity for queue")))
+
+    (testing "when queue-max-size greater than default-queue-min-threads"
+      (let [max-threads    42
+            queue-min-size (inc jetty/default-queue-min-threads)
+            {:keys [thread-pool queue]}
+            (get-thread-pool-and-queue-for-create-server max-threads
+                                                         queue-min-size)]
+        (is (= max-threads (.getMaxThreads thread-pool))
+            "Unexpected max threads for thread pool")
+        (is (= jetty/default-queue-min-threads (.getMinThreads thread-pool))
+            "Unexpected min threads for queue")
+        (is (= jetty/default-queue-min-threads (.getCapacity queue))
+            "Unexpected initial capacity for queue")
+        (is (= queue-min-size (.getMaxCapacity queue))
+            "Unexpected max capacity for queue")))))
+
+(defn get-responses
+  [num-requests port]
+  (doall (for [_ (range num-requests)]
+           (future
+             (http-sync/get
+               (str "http://localhost:"
+                    port
+                    "/hello")
+               {:as :text})))))
+
+(deftest queue-max-size-tests
```

Comment:
FWIW, I tried out the following modifications to the previous test and they all passed locally across 20+ iterations.  This approach reduces `max-threads` to the bare minimum needed for the default number of selectors and acceptors and sets `queue-max-size` to the same for the expected failure case.  The number of requests is reduced to be just above this size.

```clj
(deftest queue-max-size-tests
   (testing "queue-max-size"
     (let [response-body  "hello world"
           app            (fn [_]
                            (Thread/sleep 2000)
                            (-> response-body
                            (rr/response)
                            (rr/status 200)
                            (rr/content-type "text/plain")
                            (rr/charset "UTF-8")))
           ;; Boot Jetty with mininum number of threads needed in order to
           ;; reduce the number of requests needed to exhaust the queue for
           ;; the low queue-max-size test case.
           max-threads (jetty-config/calculate-required-threads
                         {:port 0}
                         (ks/num-cpus))
           ;; For the low queue-max-size case, need to generate just enough
           ;; concurrent requests to hit the queue limit and for additional
           ;; requests to be dropped
           num-requests   (+ 2 max-threads)]
           (testing "of default (infinite) size doesn't lead to any request fails"
             (with-test-webserver-and-config
               app port {:max-threads max-threads
                         :shutdown-timeout-seconds 0}
               (let [responses (get-responses num-requests port)]
                 (doseq [response responses]
                   (is (= 200 (:status @response))
                     "Unsuccessful response status for one request")
                   (is (= response-body (:body @response))
                     "Unsuccessful response body for one request")))))
           (testing "of sufficiently low queue-max-size causes some requests to fail"
             (with-test-webserver-and-config
               app port {:max-threads max-threads
                         :queue-max-size max-threads
                         :shutdown-timeout-seconds 0}
               (let [responses (doall (get-responses num-requests port))]
                 (is (some #(try
                   @%
                   false
                   (catch ExecutionException e
                     (instance? ConnectionClosedException
                       (.. e getCause))))
                           responses)
                     "Didn't encounter connection close for any requests")))))))
```

The critical difference this approach has is that the failure test would try to force the queue to overflow because all worker threads would be exhausted fulfilling requests.  With the previous approach, the worker threads would not be exhausted and so having the queue be overflowing would be more timing-dependent.

The above test would also run more slowly in that the reduced number of worker threads to handle the incoming requests would cause request processing on the worker threads to be done more serially.

In any event, I still don't like the race conditions baked into these tests and am not sure we'd want to introduce the fragility into the standard test suite.